### ISSUE TITLE
[BUG] org.opensearch.action.support.clustermanager.TransportClusterManagerNodeActionTests.testDontAllowSwitchingCompatibilityModeForClusterWithMultipleVersions is flaky

### ISSUE THREAD

**Issue by reta** [opensearch-project/OpenSearch#13240]
### Describe the bug

The test case `org.opensearch.action.support.clustermanager.TransportClusterManagerNodeActionTests.testDontAllowSwitchingCompatibilityModeForClusterWithMultipleVersions` is flaky:

```
java.lang.AssertionError: current nodes in the cluster must be of different versions
	at __randomizedtesting.SeedInfo.seed([3C75F620F4F6F25A:8ECD481729A878B6]:0)
	at org.opensearch.action.support.clustermanager.TransportClusterManagerNodeActionTests.testDontAllowSwitchingCompatibilityModeForClusterWithMultipleVersions(TransportClusterManagerNodeActionTests.java:832)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1750)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:938)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:974)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:988)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
	at org.apache.lucene.tests.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:48)
	at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
	at org.apache.lucene.tests.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:45)
	at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
	at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:368)
	at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:817)
	at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:468)
	at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:947)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:832)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:883)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:894)
	at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at org.apache.lucene.tests.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:38)
	at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
	at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at org.apache.lucene.tests.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
	at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
	at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
	at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
	at org.apache.lucene.tests.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:47)
	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:368)
	at java.base/java.lang.Thread.run(Thread.java:1583)
```

### Related component

Cluster Manager

### To Reproduce

```
./gradlew ':server:test' --tests "org.opensearch.action.support.clustermanager.TransportClusterManagerNodeActionTests.testDontAllowSwitchingCompatibilityModeForClusterWithMultipleVersions" -Dtests.seed=3C75F620F4F6F25A 
```

### Expected behavior

The test must always pass

### Additional Details

**Plugins**
Please list all plugins currently enabled.

**Screenshots**
Standard

**Host/Environment (please complete the following information):**
 - CI

**Additional context**
- https://build.ci.opensearch.org/job/gradle-check/37125/testReport/junit/org.opensearch.action.support.clustermanager/TransportClusterManagerNodeActionTests/testDontAllowSwitchingCompatibilityModeForClusterWithMultipleVersions/

**Comment by mch2** [comment]
Seed `-Dtests.seed=1E9A90DB5E1BB3C9` fails 100% of the time - @ltaragi can you please take a look here asap? I see this was introduced today.

**Comment by peternied** [comment]
[Triage - attendees [1](https://github.com/vikasvb90/) [2](https://github.com/reta/) [3](https://github.com/scrawfor99) [4](https://github.com/jed326) [5](https://github.com/kotwanikunal) [6](https://github.com/peternied) [7](https://github.com/lukas-vlcek/)]
@reta Thanks for filing

**Comment by reta** [comment]
Could be related to https://github.com/opensearch-project/OpenSearch/issues/13250

**Comment by ltaragi** [comment]
@mch2 fixed the faulty assertion in this test here #13281
